nemo/collections/common/prompts/formatter.py

```diff
@@ -255,6 +262,11 @@ def encode_dialog(self, turns: list[dict]) -> dict[str, torch.Tensor]:
         turn_token_counts = []
         turn_mask_values = []
 
+        if self.INSERT_BOS:
```

Comment:
What if there are custom BOS and EOS tokens that are not part of tokenizer.bos/eos? I think some models have multiple tokens to halt / their EOS os more like EOT (llama). Manually put it inside the prompt ?

Reply:
Yes, every custom/special token is normally expressed as a regular string in the prompt template. INSERT_BOS/EOS is specifically for LLMs that use sentencepiece tokenizers which have standard BOS/EOS that don't normally exist in the tokenizer's vocab (i.e. no string input generates them and you must insert them programmatically), see: https://github.com/google/sentencepiece/issues/102#issuecomment-397150427

Reply:
Ok this needs to be clearly stated in the code itself (maybe I missed it)